src/lib.rs

```diff
@@ -553,35 +567,51 @@ where
     Ok(ro)
 }
 
-/// Derive the record commitment corresponding to collected fee from a list of
-/// transaction notes (within a block). The result is a record commitment to be
-/// inserted in the ledger. This function is intended to be called by nodes
-/// validating blocks.
+/// Derive a list of record commitments corresponding to fees from a list of
+/// transaction notes (within a block). The result is a vector of record
+/// commitment which may be further aggregated into a single one by the caller.
+/// This function is intended to be called by nodes validating blocks.
 ///
 /// * `txns` - List of verified transaction notes within a block
 /// * `owner_pk` - Public key of the owner of the collected fee, usually the
 ///   block proposer's public key
 /// * `blind` - blinding factor of the record commitment
-pub fn derive_txns_fee_record(
+pub fn derive_txns_fee_records(
     txns: &[TransactionNote],
-    owner_pk: UserPubKey,
-    blind: BlindFactor,
-) -> Result<RecordCommitment, TxnApiError> {
+    owner_pks: &[UserPubKey],
+    blinds: &[BlindFactor],
+) -> Result<Vec<RecordCommitment>, TxnApiError> {
     if txns.is_empty() {
```

Comment:
this is not exactly what we discussed on Zulip, should be:
`pub fn derive_txn_fee_records(fee_collectors: &[UserPubKey], fees: Vec<u64>, blinds: Vec<BlindFactor>) ->Result<Vec<RecordCommitment, TxnApiError>`

and also remove `prepare_txn_fee_record` API.